src/main/java/graphql/execution/ExecutionStrategy.java

```diff
@@ -532,11 +531,11 @@ protected FieldValueInfo completeValueForList(ExecutionContext executionContext,
                             .path(indexedPath)
                             .source(value.getFetchedValue())
             );
-            fieldValueInfos.add(completeValue(executionContext, newParameters));
+            fieldValueInfos.add(CompletableFuture.supplyAsync(() -> completeValue(executionContext, newParameters)));
```

Comment:
This is exactly the same problem with threading but with a new method.

`CompletableFuture.supplyAsync` is also a threaded method that uses the standard ForkJoin executor pool to run things

This has the same problem as before - it introduces threading and dependency on a certain executor pool